### ISSUE TITLE
Bounded Word Wrap does not work with Markdown Language

### ISSUE THREAD

**Issue by HopefulLlama** [Microsoft/vscode#45653]
- VSCode Version: 1.21.0
- OS Version: Windows 10 Enterprise: Build 1703

Steps to Reproduce:

1. Have settings to enable bounded word wrap such as:
```
    ...
    "editor.wordWrap": "bounded",
    "editor.wordWrapColumn": 80,
    ...
```
2. Open/Create a Markdown language file
3. Write some text that exceeds `editor.wordWrapColumn`
4. Toggle word wrap on
5. See that it does not wrap at `editor.wordWrapColumn`
6. Shrink the VS Code window such that the window is thinner than text length
7. See that it wraps at window bound
8. Maximise window 
9. Change language (to Java for example)
10. Repeat, and see that word wrapping works on `editor.wordWrapColumn`

Does this issue occur when all extensions are disabled?: Yes/No
Yes

**Comment by yume-chan** [comment]
Markdown language has it's own default value for `editor.wordWrap`, you can find it under the `[markdown]` key.

Visual Studio Code supports you to set configuration to specified languages.

**Comment by HopefulLlama** [comment]
I was under the impression that *not using* the `[languageId]` syntax applies to settings to all languages. Is this incorrect?

**Comment by yume-chan** [comment]
@HammyHavoc Yes, but `[languageId]` block has **higher priority**, and there **is** a default value of `editor.wordWrap` defined for `[markdown]`.

Actually you made me thinking about is it not intuitive that a default value overriding a user value. Maybe the priority should be `user [languageId] block` -> `user settings` -> `default [languageId] block` -> `default settings`.

**Comment by alexandrudima** [comment]
Today, we ship with:

```js
  "editor.wordWrap": "off",
  "editor.wordWrapColumn": 80,
  "[markdown]": {
    "editor.wordWrap": "on",
  }
```

In order to change the settings for all files, including markdown you must write:

```js
  "editor.wordWrap": "bounded",
  "[markdown]": {
    "editor.wordWrap": "bounded",
  }
```

---

**Comment by sandy081** [comment]
@yume-chan Interesting idea. But what would be priority when presence of workspace settings. Does the `workspace settings` override `user [langId] settings` too?

**Comment by cesardramirez** [comment]
File > Preferences > Settings

```
{
    ...
    "editor.wordWrap": "bounded",
    "editor.wordWrapColumn": 100,
    "[markdown]": {
        "editor.wordWrap": "bounded",
    }
}
```
